### PR TITLE
refactor: remove BasePlugin mod

### PR DESCRIPTION
**Pull request by deryrahman**
Tracker:
- [x] Clean up plugin repo where accepting base plugin
- [x] Remove grpc go-plugin implementation on `plugin/v1beta1`
- [x] Deprecate base proto endpoint on proton https://github.com/odpf/proton/pull/209
- [x] Remove relevant test for base plugin replace it with YamlMod
- [x] Remove BasePlugin and CLIMod on bq2bq https://github.com/odpf/transformers/pull/58

### CODE REVIEW

**Comment by deryrahman** [comment]
Now, the flow of loading plugin will begin on reading yaml plugin first and then binary plugin, since the binary plugin require information eg. plugin name, from yaml.

Later on, we can do packaging the yaml+binary in one release, so that reading the plugin can be done in one shot. @smarch-int

File: models/plugin.go [hunk 1]
```diff
@@ -516,31 +480,33 @@ func (s *registeredPlugins) add(baseMod BasePlugin, drMod DependencyResolverMod,
...
+	}
+
 	switch info.PluginType {
 	case PluginTypeTask:
 	case PluginTypeHook:
 	default:
 		return ErrUnsupportedPlugin
 	}
 
-	isCandidatePluginYaml := yamlMod != nil
```
Comment (smarch-int) [inline models/plugin.go:526]: this totally removes check for duplicate plugins

File: plugin/plugin.go [hunk 2]
```diff
@@ -3,99 +3,44 @@ package plugin
...
-		}
-		pluginLogger.Debug("plugin ready: ", baseInfo.Name)
-	}
-
-	// first fetch binary plugins, then add yaml plugin for which binary is not added earlier
-	discoveredYamlPlugins := DiscoverPluginsGivenFilePattern(pluginLogger, yaml.Prefix, yaml.Suffix)
+	// fetch yaml plugins first, it holds detailed information about the plugin
+	discoveredYamlPlugins := discoverPluginsGivenFilePattern(pluginLogger, yaml.Prefix, yaml.Suffix)
 	pluginLogger.Debug(fmt.Sprintf("discovering yaml   plugins(%d)...", len(discoveredYamlPlugins)))
 	yaml.Init(models.PluginRegistry, discoveredYamlPlugins, pluginLogger)
```
Comment (smarch-int) [inline plugin/plugin.go:22]: handle error. earlier yamlmod was optional.

File: plugin/yaml/plugin.go [hunk 3]
```diff
@@ -107,15 +107,15 @@ func NewPluginSpec(pluginPath string) (*PluginSpec, error) {
...
 		yamlPluginSpec, err := NewPluginSpec(yamlPluginPath)
 		if err != nil {
 			pluginLogger.Error(fmt.Sprintf("plugin Init: %s", yamlPluginPath), err)
 			continue
 		}
-		pluginInfo, _ := yamlPluginSpec.PluginInfo()
+		pluginInfo := yamlPluginSpec.PluginInfo()
 		if err := pluginsRepo.AddYaml(yamlPluginSpec); err != nil {
 			pluginLogger.Error(fmt.Sprintf("PluginRegistry.Add: %s", yamlPluginPath), err)
 			continue
```
Comment (smarch-int) [inline plugin/yaml/plugin.go:121]: dont skip errors as yamlmod is compulsory

File: plugin/yaml/tests/sample_plugin_without_version.yaml [hunk 4]
```diff
@@ -3,7 +3,6 @@ description: Testing
 plugintype: task
 pluginmods:
   - cli
-pluginversion: ""
```
Comment (smarch-int) [inline plugin/yaml/tests/sample_plugin_without_version.yaml:6]: why ?

File: models/plugin.go (hunk 1 above)
Comment (smarch-int) [reply models/plugin.go:526]: refer https://github.com/odpf/optimus/pull/567/files#diff-a252f8221d35378ca0f6616ad8512f4bd5ee6ef60e724d1399cd7dfe1ae852f7L545

File: plugin/plugin.go (hunk 2 above)
Comment (deryrahman) [reply plugin/plugin.go:22]: it's already coverd in binary Init

File: plugin/yaml/tests/sample_plugin_without_version.yaml (hunk 4 above)
Comment (deryrahman) [reply plugin/yaml/tests/sample_plugin_without_version.yaml:6]: renamed to `sample_plugin_empty_version`

File: plugin/binary/plugin.go [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+		pluginClient := plugin.NewClient(&plugin.ClientConfig{
+			HandshakeConfig:  dependencyresolver.Handshake,
+			Plugins:          pluginMap,
+			Cmd:              exec.Command(pluginPath, args...),
+			Managed:          true,
+			AllowedProtocols: []plugin.Protocol{plugin.ProtocolGRPC},
+			Logger:           pluginLogger,
+		})
+
+		pluginName := getNameFromPluginPath(pluginPath)
```
[the remark this answers is not in the document]
Comment (deryrahman) [reply plugin/binary/plugin.go:39]: can add new method, getName to retrieve the binary plugin name

File: models/plugin.go (hunk 1 above)
Comment (deryrahman) [reply models/plugin.go:530]: adding alternative logic to achieve same result, please check

**Review by smarch-int — APPROVED** [review]
LGTM!

**Pull request merged**